Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/PropertyBean.java

```diff
@@ -0,0 +1,139 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY 
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES 
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND 
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT 
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS 
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ */
+package gov.hhs.fha.nhinc.admingui.managed;
+
+import gov.hhs.fha.nhinc.properties.PropertyAccessException;
+import gov.hhs.fha.nhinc.properties.PropertyAccessor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import javax.faces.application.FacesMessage;
+import javax.faces.bean.ManagedBean;
+import javax.faces.bean.ViewScoped;
+import javax.faces.context.FacesContext;
+import org.apache.log4j.Logger;
+import org.primefaces.component.datatable.DataTable;
+import org.primefaces.event.CellEditEvent;
+
+/**
+ *
+ * @author jassmit
+ */
+@ManagedBean(name = "propertyBean")
+@ViewScoped
+public class PropertyBean {
+
+    private List<PropValue> properties;
+    
+    private static final Logger LOG = Logger.getLogger(PropertyBean.class);
+
+    public PropertyBean() {
+        setProperties();
+    }
+
+    public List<PropValue> getProperties() {
+        return properties;
+    }
+
+    public void refresh() {
+        setProperties();
+    }
+
+    public void onValueEdit(CellEditEvent event) {
+        DataTable dataTable = (DataTable) event.getSource();
+        PropValue selectedProp = (PropValue) dataTable.getRowData();
+        String oldValue = (String) event.getOldValue();
+        String newValue = (String) event.getNewValue();
+
+        try {
+            PropertyAccessor.getInstance().setProperty("gateway", selectedProp.key, newValue);
+            FacesContext.getCurrentInstance().addMessage(null,
+                    new FacesMessage(FacesMessage.SEVERITY_INFO, "INFO", "Property value changed for " + selectedProp.key
+                            + " from " + oldValue + " to " + newValue + "."));
+        } catch (PropertyAccessException ex) {
+            LOG.warn("Unable to update property: " + selectedProp.key);
+            FacesContext.getCurrentInstance().addMessage(null,
+                    new FacesMessage(FacesMessage.SEVERITY_WARN, "WARN", "Unable to set property value: " + selectedProp.key
+                            + " from " + oldValue + " to " + newValue + "."));
+        }
+    }
+
+    private void setProperties() {
+        properties = new ArrayList<PropValue>();
+        try {
+            Properties props = PropertyAccessor.getInstance().getProperties("gateway");
+            
+            if (props != null) {
+                for (Object key : props.keySet()) {
```

Comment:
the key set returns the type Set<Object>, I cast on each iteration.  can't cache on the set.